src/modules/course/schema/enrolledCourse.schema.ts

```diff
@@ -0,0 +1,41 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+import * as mongoose from 'mongoose';
+
+export type EnrolledCourseDocument = EnrolledCourse & Document;
+
+interface video {
+  num: number;
+  timestamp?: Date;
+}
+
+@Schema()
+export class EnrolledCourse {
+  @Prop({})
+  eId: mongoose.Schema.Types.ObjectId;
+
+  @Prop({ default: [false, false, false, false, false] })
+  videos_watched: boolean[];
+
+  @Prop({ default: [false, false, false, false, false] })
```

Comment:
Why this 5 length boolean array?
the number of videos will not be 5 everytime.
Use schema hooks, 

like `EnrolledCourses.pre('save', () => {
videos_watched = new Array(10);
}
`
and use .fill to fill the size with false.

Ref: https://mongoosejs.com/docs/middleware.html#pre

Reply:
Ok

Reply:
ok will do this

Reply:
default: [].fill(false)

can i do this instead of the pre commit hook?
Same result will be obtained i guess